### PR TITLE
Fix time remaining wrapping back to zero when map is longer than hour

### LINKED ISSUE #2466: Time remaining wraps back around to zero when a map is an hour long or longer
This can be seen happening with the map [Hatsune Miku Remembers 10,000 Digits of Pi](https://osu.ppy.sh/beatmapsets/355637). osu!lazer starts off saying there's only 8 minutes left in the song, but after those 8 minutes are up, will say there's 59 minutes left. Similarly, after 59 minutes have passed, lazer will say no time has passed.
osu _2018-04-28_00-21-13
osu _2018-04-28_00-25-30

### PR DESCRIPTION
**Pull request by Qapples**
This resolves #2466

### CODE REVIEW

File: osu.Game/Screens/Play/SongProgressInfo.cs [hunk 1]
```diff
@@ -85,8 +85,8 @@ protected override void Update()
 
             if (currentSecond != previousSecond && songCurrentTime < songLength)
             {
-                timeCurrent.Text = TimeSpan.FromSeconds(currentSecond).ToString(songCurrentTime < 0 ? @"\-m\:ss" : @"m\:ss");
-                timeLeft.Text = TimeSpan.FromMilliseconds(endTime - AudioClock.CurrentTime).ToString(@"\-m\:ss");
+                timeCurrent.Text = TimeSpan.FromSeconds(currentSecond).ToString(songCurrentTime < 0 ? @"\-h\:mm\:ss" : @"h\:mm\:ss");
```
Comment (swoolcock) [inline osu.Game/Screens/Play/SongProgressInfo.cs]: I think you should only be showing hours if the map goes for over an hour.  `0:02:30` looks horrible, when the majority of maps are going to be under 10 minutes.

**Review by swoolcock — CHANGES_REQUESTED** [review]

File: osu.Game/Screens/Play/SongProgressInfo.cs (hunk 1 above)
Comment (Qapples) [reply osu.Game/Screens/Play/SongProgressInfo.cs]: 0:02:30 looks normal and fine to me, I've seen the h:mm:ss format in a lot of timers in and out of video games. But, I see your point.

Comment (swoolcock) [reply osu.Game/Screens/Play/SongProgressInfo.cs]: That initial `0:0` will be redundant for anything but marathons and unrankables.  Sometimes less information is clearer.

Comment (Qapples) [reply osu.Game/Screens/Play/SongProgressInfo.cs]: Alright, understood.

Comment (peppy) [reply osu.Game/Screens/Play/SongProgressInfo.cs]: Personally i'd prefer it to keep counting in minutes, not hours.

**Comment by peppy** [comment]
Closing in favour of #2487...?

**Pull request closed without merging**